### DISCUSSION TITLE
accuracy of math.pi

### DISCUSSION THREAD

**Discussion by Gerhard34** [micropython/micropython#16479] [Libraries & Drivers] [answered]
I have a very simple script which show pi for 15 digits. I wounder about the result: 3.14159274101257. 
According to other calculations (e.g. Excel) it should be: 3.14159265358979

Could anyone pls. help to clarify the different results?
thx Gerhard

```py
import math
d = math.pi
print(f'{d:.15}')
```

Platform: MicroPython v1.24.1 on 2024-11-29; ESP32C3 module with ESP32C3

**Comment by robert-hh** [comment, marked as answer]
Esp32c3 uses single float. So, only 6-7 digits are significant.

**Comment by sosi-deadeye** [comment]
3.14 is accurate enough for the most calculations.

**Comment by Gerhard34** [comment]
Thank you @robert-hh for you explanation!

@sosi-deadeye: I agree that this might be sufitient for most cases, but I'm running GPS calculations with the [Haversine formula](https://en.wikipedia.org/wiki/Haversine_formula) on short distance between two GPS points. This requires a more presice calculation.

**Comment by rcolistete** [comment]
Few MicroPython firmwares have FP64 (double precision for floats), like Pyboard :
https://micropython.org/download/PYBV11/
Some MicroPython boards can have firmware compiled with FP64. Like ESP32 :
https://github.com/rcolistete/ulab_samples

Other solution :
Announcing DecimalNumber: floating point arithmetic with arbitrary precision for micropython
https://forum.micropython.org/viewtopic.php?t=11375

**Reply by peterhinch** [reply]
To clarify re Pyboards:

The Pyboard V1.1 has 32-bit hardware FP, but it can have firmware compiled with FP64. This is available as a precompiled build. The Pyboard D SF6W has 64-bit hardware FP.

**Reply by scruss** [reply]
There are also unofficial FP64 builds for the RP2040, but they are tremendously slow.

DecimalNumber is wonderful, but even slower than all that.
